src/hera/workflow.py

```diff
@@ -143,23 +143,19 @@ def __init__(
             parallelism=self.parallelism,
         )
 
+        self.spec = IoArgoprojWorkflowV1alpha1WorkflowSpec(
+            templates=[self.template],
+            entrypoint=self.name,
+            volumes=[],
+            volume_claim_templates=[],
+            parallelism=self.parallelism,
+        )
+
         if self.workflow_template_ref:
```

Comment:
Makes this bit a little bit simpler to read (no code dupe): we set the spec with the template by default and only override with a ref if it's set.